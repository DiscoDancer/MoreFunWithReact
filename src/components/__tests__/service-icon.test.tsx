import * as React from "react";
import ServiceIcon from "../service-icon.component";
import * as renderer from 'react-test-renderer';
import ServiceIconModel from "../../models/service-icon.model";

test('ServiceIcon passive by props snapshot test', () => {

    const model = new ServiceIconModel(
        "/icon.jpg",
        "/active_icon.jpg",
        "name",
        "description"
    );

    const component = renderer.create(
        <ServiceIcon model={model} isActive={false}/>,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('ServiceIcon active by props snapshot test', () => {

    const model = new ServiceIconModel(
        "/icon.jpg",
        "/active_icon.jpg",
        "name",
        "description"
    );

    const component = renderer.create(
        <ServiceIcon model={model} isActive={true}/>,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
