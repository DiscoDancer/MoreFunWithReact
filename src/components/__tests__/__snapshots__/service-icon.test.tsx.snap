// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceIcon active by props snapshot test 1`] = `
<li
  className="col-2 service-icon active"
>
  <figure>
    <img
      alt="service icon image"
      className="passive-image"
      src="/icon.jpg"
    />
    <img
      alt="service icon image when it's active"
      className="active-image"
      src="/active_icon.jpg"
    />
    <figcaption>
      name
    </figcaption>
  </figure>
</li>
`;

exports[`ServiceIcon passive by props snapshot test 1`] = `
<li
  className="col-2 service-icon passive"
>
  <figure>
    <img
      alt="service icon image"
      className="passive-image"
      src="/icon.jpg"
    />
    <img
      alt="service icon image when it's active"
      className="active-image"
      src="/active_icon.jpg"
    />
    <figcaption>
      name
    </figcaption>
  </figure>
</li>
`;
